nltk/tag/crfsuite.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+# Natural Language Toolkit: Interface to the CRFSuite Tagger
+#
+# Copyright (C) 2001-2015 NLTK Project
+# Author: Long Duong <longdt219@gmail.com>
+# URL: <http://nltk.org/>
+# For license information, see LICENSE.TXT
+
+"""
+A module for interfacing with the CRFSuite taggers.
+"""
+from __future__ import absolute_import
+from __future__ import unicode_literals
+import unicodedata
+import re 
+from nltk.tag.api import TaggerI
+import pycrfsuite
+
+_py_crfsuite_url = 'https://pypi.python.org/pypi/python-crfsuite'
+
+
+class CRFTagger(TaggerI):
+    """
+    An interface to Python CRFSuite taggers. https://pypi.python.org/pypi/python-crfsuite
+    
+    >>> from nltk.tag.crfsuite import CRFTagger
+    >>> ct = CRFTagger()
+ 
+    >>> train_data = [[('University','Noun'), ('is','Verb'), ('a','Det'), ('good','Adj'), ('place','Noun')],
+    ... [('dog','Noun'),('eat','Verb'),('meat','Noun')]]
+    
+    >>> ct.train(train_data,'model.crf.tagger')
+    >>> ct.tag_sents([['dog','is','good'], ['Cat','eat','meat']])
+    [[('dog', 'Noun'), ('is', 'Verb'), ('good', 'Adj')], [('Cat', 'Noun'), ('eat', 'Verb'), ('meat', 'Noun')]]
+    
+    >>> gold_sentences = [[('dog','Noun'),('is','Verb'),('good','Adj')] , [('Cat','Noun'),('eat','Verb'), ('meat','Noun')]] 
+    >>> ct.evaluate(gold_sentences) 
+    1.0
+    
+    Setting learned model file  
+    >>> ct = CRFTagger() 
+    >>> ct.set_model_file('model.crf.tagger')
+    >>> ct.evaluate(gold_sentences)
+    1.0
+    
+    """
+    
+    def __init__(self, file_path=''):
+               
+        self._model_file = ''
+    
+
+    def set_model_file(self, model_file):
+        self._model_file = model_file
+            
+    def _get_features(self,data):
+        """
+        Extract basic features about this word including 
+             - Current Word 
+             - Is Capitalized ?
+             - Has Punctuation ?
+             - Has Number ?
+             - Suffixes up to length 3 
+        :return : a string which contains the features   
+        
+        """ 
+        feature_list = []  
+        # Capitalization 
+        if data[0].isupper():
+            feature_list.append('CAPITALIZATION')
+        
+        # Number 
+        pattern = re.compile('\\d')
```

Comment:
It makes no sense to compile a pattern every time we extract a feature. Actually, it should be faster if you just remove .compile call because there is an internal regexp cache in re module. 

`r'\d'` looks better than `'\\d'`.